cluster-agent/controllers/argoproj.io/application_controller.go

```diff
@@ -114,6 +118,13 @@ func (r *ApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			applicationState.Revision = db.TruncateVarchar(app.Status.Sync.Revision, db.ApplicationStateRevisionLength)
 			sanitizeHealthAndStatus(applicationState)
 
+			// Get the list of resources created by deployment and convert it into a compressed YAML string.
+			var err error
+			applicationState.Resources, err = CompressResourceData(app.Status.Resources)
+			if err != nil {
+				log.Error(err, "unable to compress resource data into byte array.")
```

Comment:
Think you can `return ctrl.Result{}, err` here, unless here is a reason not to.

Reply:
Changed it, my thinking here was that we can continue with deployment even though Resource field is not set, just like it is at present.